components/lib/VCarousel.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+  <div class="slider">
+    <transition-group :name="'slide-' + transitionName" tag="div">
+      <div :key="currentSlide" ref="carousel" class="slide">
+        <slot />
+      </div>
+    </transition-group>
+    <v-stack class="icons" gap="2rem" justify="center">
+      <v-image path="icons/arrow-left.svg" @click.stop="changeSlide(-1)" />
+      <v-image path="icons/arrow-right.svg" @click.stop="changeSlide(1)" />
+    </v-stack>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'VCarousel',
+  data() {
+    return {
+      currentSlide: 0,
+      transitionName: '',
+    }
+  },
+  computed: {
+    totalSlides() {
+      return this.$refs.carousel.children.length
+    },
+  },
+  mounted() {
+    this.setupSlides()
+  },
+  methods: {
+    enterPreviousSlide() {
+      this.transitionName = 'prev'
+      if (this.currentSlide - 1 >= 0) {
+        this.currentSlide--
+      } else {
+        this.currentSlide = this.totalSlides - 1
+      }
+    },
+    enterNextSlide() {
+      this.transitionName = 'next'
+      if (this.currentSlide + 1 < this.totalSlides) {
+        this.currentSlide++
+      } else {
+        this.currentSlide = 0
+      }
+    },
+    changeSlide(position) {
+      if (position === 1) {
+        this.enterNextSlide()
+      } else {
+        this.enterPreviousSlide()
+      }
+      // Using nextTick to update classes after transition event is fired
+      // Updating carousel classes before transition event does nothing
+      this.$nextTick(() => {
+        this.setActiveSlide()
+      })
+    },
+    setupSlides() {
+      this.$refs.carousel.children.forEach((child, index) => {
+        child.classList.add('carousel-item')
+        if (index === 0) {
+          child.classList.add('active-item')
+        }
+      })
+    },
+    setActiveSlide() {
+      this.$refs.carousel.children.forEach((child, index) => {
+        child.classList.add('carousel-item') // Added this because it was getting removed on update. Not sure why
```

Comment:
@punaifang I need to add this on every slide change because the entire classList is getting reset, not sure why